core/shared/platform/common/posix/posix_socket.c

```diff
@@ -318,6 +333,7 @@ os_socket_inet_network(bool is_ipv4, const char *cp, bh_ip_addr_buffer_t *out)
         /* Note: ntohl(INADDR_NONE) == INADDR_NONE */
         out->ipv4 = ntohl(out->ipv4);
     }
+#ifdef IPPROTO_IPV6
     else {
```

Comment:
Do we handle it correctly? I think if `is_ipv4` is set to `false` and ipv6 is not supported, we should probably return an error instead (I think right now we return `BHT_OK`).

Reply:
You are right, changed this to return `BHT_ERROR` and set `EAFNOSUPPORT` as the `errno` where applicable.